sycamore/transforms/query.py

```diff
@@ -0,0 +1,57 @@
+from abc import abstractmethod, ABC
+from typing import Any
+
+from opensearchpy import OpenSearch
+from ray.data import Dataset
+
+from sycamore.data import OpenSearchQueryResult, Element, OpenSearchQuery
+from sycamore.plan_nodes import Node, NonCPUUser, NonGPUUser, Transform
+from sycamore.utils.generate_ray_func import generate_map_function
+
+
+class QueryExecutor(ABC):
+    @abstractmethod
+    def query(self, query: Any) -> Any:
+        pass
+
+    def __call__(self, query: Any) -> Any:
+        return self.query(query)
+
+
+class OpenSearchQueryExecutor(QueryExecutor):
+    def __init__(self, os_client_args: dict) -> None:
+        super().__init__()
+        self._os_client_args = os_client_args
+
+    def query(self, query: OpenSearchQuery) -> OpenSearchQueryResult:
+        client = OpenSearch(**self._os_client_args)
+
+        os_result = client.transport.perform_request(
+            "POST",
+            url=f"/{query['index']}/_search",
+            params=query.get("params", None),
+            headers=query.get("headers", None),
+            body=query["query"],
+        )
+
+        result = OpenSearchQueryResult()
+        result.result = os_result
+        result.hits = [Element(hit["_source"]) for hit in os_result["hits"]["hits"]]
```

Comment:
maybe keep the scores? idk if that's necessary tbh but seems like something that would be nice to have in some cases (at the very least debugging)

Reply:
^We store the entire result above so should be able to access everything. I'll verify

Reply:
Had a bug, fixed and verified. Tests with evaluation will cover it as well